### PR TITLE
85 unified editing of data

### PR DESCRIPTION
**Pull request by YechiamW3**
Note: still left to handle username change; I've specifically left it because a small discussion is needed: because we want to (at least) rely on username for the profile page routing, changing it means a different route for the profile - so when the user changes the username we'll need to also redirect to the new page. Do we want to keep previous usernames' pages and redirect via their link to the new username page? I can also just add a cache tag to the previous username's tag to redirect to the new username; but that can be done efficiently only to the last-used username. Or to be more exact, doing it for _all_ the previous usernames will be cumbersome, and I don't think we need to handle such case.

Anyway, redirect to the new username page - a must. Updating the currently previous username's cache to redirect to the new one - bonus. Should I add it?

### CODE REVIEW

**Comment by YechiamW3** [comment]
Also @Aradon267, I've added a note on what we discussed regarding validation, let me know if I can safely delete it or keep it for now:
https://github.com/magshimim-next/cv-next/pull/204/files#diff-275c1e4152a47eb79a69accd42fa86187b6dced472f98a5ebce27dca1af59283R91-R97

**Comment by Aradon267** [comment]
I'll remove the constraint and yes this can be removed

**Comment by Aradon267** [comment]
And about what you said about the username: currently, the links are based on getting the path from the URL(the UUID) and then fetching all data based on that. The switch that the URL would have the username is happening on #202 so maybe we can leave it like that and the links will still be based on the usernames?

**Comment by Aradon267** [comment]
Several things that I'm not sure where to put:
1. I needed to run `npm i react-hook-form` first
2. Going on a different profile gave me my data:
image

3. the work status shouldn't be with _ in it, it was like that for the supabase enum

**Comment by YechiamW3** [comment]
> Several things that I'm not sure where to put:
> 
> 1. I needed to run `npm i react-hook-form` first
> 2. Going on a different profile gave me my data:
>    image
> 3. the work status shouldn't be with _ in it, it was like that for the supabase enum

1. You needed to do 'npm i', because a new package was added (and the package.json was modified accordingly). You had to do specifically `npm i react-hook-form` even after doing `npm i`?
2. Welp, will check
3. Right, forgot to check work status. Will fix, thanks

**Comment by YechiamW3** [comment]
> And about what you said about the username: currently, the links are based on getting the path from the URL(the UUID) and then fetching all data based on that. The switch that the URL would have the username is happening on #202 so maybe we can leave it like that and the links will still be based on the usernames?

OK, I'm leaving the TODO so that we won't forget to handle it.

File: cv_next/app/profile/[profileId]/components/profilePersonalData.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    if (userData && user.id === userData.id) {
+      setIsCurrentUser(true);
+    } else {
+      setIsCurrentUser(false);
+    }
+  }, [user.id, userData]);
+
+  const changePfp = () => {
+    //To be implemented
+  };
```
Comment (Aradon267) [inline cv_next/app/profile/[profileId]/components/profilePersonalData.tsx:32]: Maybe there should be like a "Coming soon" thing when the user clicks on the option itself?

Comment (YechiamW3) [reply cv_next/app/profile/[profileId]/components/profilePersonalData.tsx:32]: I do think we need to handle it as part of V1, no?
If not, then sure. lmk.

Comment (YechiamW3) [reply cv_next/app/profile/[profileId]/components/profilePersonalData.tsx:32]: tbf if we don't have it as part of V1 I can simply remove this edit option.

Comment (Aradon267) [reply cv_next/app/profile/[profileId]/components/profilePersonalData.tsx:32]: It's just that currently the way we handle the images is just via the links from gmail(we don't upload any image or something to our own bucket)

Comment (YechiamW3) [reply cv_next/app/profile/[profileId]/components/profilePersonalData.tsx:32]: So, we currently don't have the means to edit the image, correct?
If not, I'll remove the edit option, and we can add it as an issue to V2.

Comment (Aradon267) [reply cv_next/app/profile/[profileId]/components/profilePersonalData.tsx:32]: Exactly, we don't
I'll add it to V2

**Pull request merged**